gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/tasks/LaunchConfigTask.java

```diff
@@ -0,0 +1,234 @@
+import org.immutables.value.Value;
+
+public class LaunchConfigTask extends DefaultTask {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper(new YAMLFactory());
+    private static final List<String> java8gcLoggingOptions = ImmutableList.of(
+            "-XX:+PrintGCDateStamps",
+            "-XX:+PrintGCDetails",
+            "-XX:-TraceClassUnloading",
+            "-XX:+UseGCLogFileRotation",
+            "-XX:GCLogFileSize=10M",
+            "-XX:NumberOfGCLogFiles=10",
+            "-Xloggc:var/log/gc-%t-%p.log",
+            "-verbose:gc"
+    );
+
+    private static final List<String> alwaysOnJvmOptions = ImmutableList.of(
+            "-XX:+CrashOnOutOfMemoryError",
+            "-Djava.io.tmpdir=var/data/tmp",
+            "-XX:ErrorFile=var/log/hs_err_pid%p.log",
+            // Set DNS cache TTL to 20s to account for systems such as RDS and other
+            // AWS-managed systems that modify DNS records on failover.
+            "-Dsun.net.inetaddr.ttl=20"
+    );
+
+    // Reduce memory usage for some versions of glibc.
+    // Default value is 8 * CORES.
+    // See https://issues.apache.org/jira/browse/HADOOP-7154
+    public static final Map<String, String> defaultEnvironment = ImmutableMap.of("MALLOC_ARENA_MAX", "4");
+
+    private Property<String> mainClass = getProject().getObjects().property(String.class);
+    private Property<String> serviceName = getProject().getObjects().property(String.class);
+    private Property<GcProfile> gc = getProject().getObjects().property(GcProfile.class);
+    private Property<Boolean> addJava8GcLogging = getProject().getObjects().property(Boolean.class);
+    private Property<String> javaHome = getProject().getObjects().property(String.class);
+    private RegularFileProperty staticLauncher = getProject().getObjects().fileProperty();
+    private RegularFileProperty checkLauncher = getProject().getObjects().fileProperty();
+    private ListProperty<String> args = getProject().getObjects().listProperty(String.class).empty();
+    private ListProperty<String> checkArgs = getProject().getObjects().listProperty(String.class).empty();
+    private ListProperty<String> defaultJvmOpts = getProject().getObjects().listProperty(String.class).empty();
+    private MapProperty<String, String> env = getProject().getObjects().mapProperty(String.class, String.class).empty();
+
+    private FileCollection classpath;
+
+    public LaunchConfigTask() {
+        staticLauncher.set(new File(getProject().getBuildDir(), "scripts/launcher-static.yml"));
+        checkLauncher.set(new File(getProject().getBuildDir(), "scripts/launcher-check.yml"));
+    }
+
+    @Input
+    public final Property<String> getMainClass() {
+        return mainClass;
+    }
+
+    @Input
+    public final Property<String> getServiceName() {
+        return serviceName;
+    }
+
+    public final Property<GcProfile> getGc() {
+        return gc;
+    }
+
+    // HACKHACK Property<GcProfile> failed to serialise
```

Comment:
Is this still true after adding Serializabel to GcProfile? Perhaps it needs a serialVersionUid?

Reply:
yup, it might be because its just an interface?